src/librustc_trans/adt.rs

```diff
@@ -87,6 +87,14 @@ impl MaybeSizedValue {
     pub fn has_meta(&self) -> bool {
         !self.meta.is_null()
     }
+
+    pub fn to_args(&self) -> Vec<ValueRef> {
```

Comment:
Actually, there's only two places so I'd rather just have allocation-less logic in both of them, this might work:
```rust
let args = &[x.value, x.meta][..1 + x.has_meta() as usize];
```
You could also make a helper for "argument count".